.github/workflows/ci-asp-app.yaml

```diff
@@ -0,0 +1,133 @@
+        uses: actions/checkout@v3
+
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.3
+      
+      - name: Setup NuGet
+        uses: NuGet/setup-nuget@v1.2
+      
+      - name: Restore NuGet packages
+        run: cd ${{ inputs.BUILD_PATH }} && nuget restore ${{ inputs.PATH_TO_SOLVE_FILE }}
+      
+      - name: Download Rattus
+        uses: robinraju/release-downloader@v1.2
+        with:
+          repository: "saritasa/rattus"
+          tag: "0.2"
+          fileName: "rattus-windows-amd64.exe"
+      
+      - name: Render config file
+        run: .\rattus-windows-amd64.exe > ${{ inputs.BUILD_PATH }}/Web.Debug.config
+        env:
+          AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+          AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
+          AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+          AZURE_VAULT: ${{ inputs.AZURE_VAULT }}
+          TEMPLATE_PATH: ${{ inputs.TEMPLATE_PATH }}
+     
+      - name: Build app for release
+        run: msbuild ${{ inputs.BUILD_PATH }}/${{ inputs.CSPROJ_FILE }} /p:Configuration=Debug /p:Platform=AnyCPU /t:WebPublish /p:WebPublishMethod=FileSystem /p:DeleteExistingFiles=True /p:publishUrl=${{ inputs.BUILD_FOLDER }}/${{ inputs.APP_NAME }}-${{ inputs.environment }}
+      
+      - name: Upload artifact for deployment job
+        uses: actions/upload-artifact@v2
+        with:
+          name: ASP-app
+          path: ${{ inputs.BUILD_PATH }}/${{ inputs.BUILD_FOLDER }}/${{ inputs.APP_NAME }}-${{ inputs.environment }}
+      
+  # Generated by Azure Depoloyment Center
+  # Docs for the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
+  # More GitHub Actions for Azure: https://github.com/Azure/actions
+  
+  # Deploying an application to azure cloud
+  deploy:
+    runs-on: windows-latest
+    needs: build
+    environment:
+      name: 'Develop'
+      url: ${{ steps.deploy-to-webapp.outputs.webapp-url }}
+    
+    steps:
+      - name: Download artifact from build job
+        uses: actions/download-artifact@v2
+        with:
+          name: ASP-app
+      
+      # slot-name should always be Production since there is only one available deployment slot
+      # A publish profile is a file that contains information and settings for deploy applications and services in Azure Cloud
+      # publish-profile is a file which is unique for each app serivce. It can be downloaded from app service's overview page on Azure portal:
+      # https://docs.microsoft.com/en-us/azure/app-service/deploy-github-actions?tabs=applevel#generate-deployment-credentials
+      
+      - name: Deploy to Azure Web App
+        id: deploy-to-webapp
+        uses: azure/webapps-deploy@v2
+        with:
+          app-name: '${{ inputs.APP_NAME }}-${{ inputs.environment }}'
+          slot-name: 'production' # deployment slot by default for new application
+          # Get publish-profile from https://keys.saritasa.cloud/cred/detail/16857/
```

Comment:
that should not be here - instead - it should be in the ci-dev.yml in the concrete repository calling this workflow.

Reply:
done